upb/msg_internal.h

```diff
@@ -100,22 +154,28 @@ struct upb_msglayout {
 typedef struct {
   upb_msglayout_field field;
   const upb_msglayout *extendee;
-  const upb_msglayout *submsg;   /* NULL for non-submessage fields. */
+  upb_msglayout_sub sub;   /* NULL unless submessage or proto2 enum */
 } upb_msglayout_ext;
 
+typedef struct {
```

Comment:
here you do `typedef struct { ...} name;` early in the patch you do `struct Foo`.  Why the difference?